addon/selection/active-line.js

```diff
@@ -33,6 +38,15 @@
     }
   });
 
+  CodeMirror.defineOption("styleActiveSelected", false, function(cm, val, old) {
+    var prev = old && old != CodeMirror.Init;
+    if (val && !prev) {
+      cm.state.maintainHighlight = true;
```

Comment:
You don't need to store anything in the state. You can just read the options when you need them using `getOption`